api/src/main/java/org/openmrs/User.java

```diff
@@ -35,14 +36,27 @@
  * key-value pairs for either quick info or display specific info that needs to be persisted (like
  * locale preferences, search options, etc)
  */
-public class User extends BaseChangeableOpenmrsMetadata implements java.io.Serializable, Attributable<User> {
-	
-	public static final long serialVersionUID = 2L;
+@Entity
+@Table(name = "users")
+@Cacheable
+@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
+/*
+*User inherits name and description properties from BaseOpenmrsMetadata but they don't exist in the users table since 
+*they are not used, because these field exist in BaseOpenmrsData and they are marked as persistent, hibernate generates 
+*a query containing these columns which of course fails to execute because of missing columns, so we introduce this 
+*dummy table containing these columns to 'please' hibernate until we fix the Class hierarchy see TRUNK-5665
+*/
+@SecondaryTable(name = "users_unused_fields")
+@AttributeOverrides({
+        @AttributeOverride(name = "name", column = @Column(table = "users_unused_fields", name = "name", insertable = false, updatable = false)),
+        @AttributeOverride(name = "description", column = @Column(table = "users_unused_fields", name = "description", insertable = false, updatable = false)) })
+public class User extends BaseOpenmrsObject implements java.io.Serializable, Attributable<User>, Auditable, Retireable {
```

Comment:
So here we really did want to end up with replacing all of this with one line:

```suggestion
public class User extends BaseOpenmrsObject implements java.io.Serializable, Attributable<User>, Auditable, Retireable {
```

Reply:
Similarly, the rest of the Git merge issues really do need to be resolved before this will compile, because otherwise you end up with things like this that Java doesn't know how to process: `<<<<<<< HEAD`